### ISSUE TITLE
Allow wildcarded field name subscriptions to CachedDataServer

### ISSUE THREAD

**Issue by davidpablocohn** [OceanDataTools/openrvdas#124]
Client should be able to specify wildcard in their subscription requests, e.g.
```
{'type':'subscribe',
 'interval':0.2,
 'fields':{'S330*':{'seconds':0}}
}
```
One nuance is that it's probably too much to ask for that the CachedDataServer add in new matching fields if they become available after the subscription request. Though perhaps there could be an additional field in the subscription request saying how often to check for new fields that may match.

**Comment by davidpablocohn** [comment]
Pull request #223 resolves the first half of this (thank you, Anshika!). Leaving the issue open pending whether it's practical to resolve the last part - picking up new matching fields as they become available.

**Comment by davidpablocohn** [comment]
Resolved with #226
